app/backend/wells/management/commands/export.py

```diff
@@ -0,0 +1,221 @@
+import csv
+import zipfile
+import os
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import models
+from django.db import connection
+
+from minio import Minio
+import xlsxwriter
+
+from gwells.settings.base import get_env_variable
+from wells.models import Well, LithologyDescription, Casing, Screen, ProductionData, Perforation
+
+# Run from command line :
+# python manage.py export
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        logger.info('starting export')
+        zip_filename = 'gwells.zip'
+        spreadsheet_filename = 'gwells.xlsx'
+        self.generate_files(zip_filename, spreadsheet_filename)
+        self.upload_files(zip_filename, spreadsheet_filename)
+        logger.info('cleaning up')
+        for filename in (zip_filename, spreadsheet_filename):
+            if os.path.exists(filename):
+                os.remove(filename)
+        logger.info('export complete')
+
+    def upload_files(self, zip_filename, spreadsheet_filename):
+        minioClient = Minio(get_env_variable('S3_HOST'),
+                            access_key=get_env_variable('S3_PUBLIC_ACCESS_KEY'),
+                            secret_key=get_env_variable('S3_PUBLIC_SECRET_KEY'),
+                            secure=True)
+        for filename in (zip_filename, spreadsheet_filename):
+            logger.info('uploading {}'.format(filename))
+            with open(filename, 'rb') as file_data:
+                file_stat = os.stat(filename)
+                # Do we need to remove the existing files 1st?
+                minioClient.put_object(get_env_variable('S3_WELL_EXPORT_BUCKET'),
+                                       filename,
+                                       file_data,
+                                       file_stat.st_size)
+
+    def export(self, workbook, gwells_zip, worksheet_name, cursor):
+        logger.info('exporting {}'.format(worksheet_name))
+        worksheet = workbook.add_worksheet(worksheet_name)
+        csv_file = '{}.csv'.format(worksheet_name)
+        if os.path.exists(csv_file):
+            os.remove(csv_file)
+        with open(csv_file, 'w') as csvfile:
+            csvwriter = csv.writer(csvfile, dialect='excel')
+
+            values = []
+            # Write the headings
+            for index, field in enumerate(cursor.description):
+                worksheet.write(0, index, '{}'.format(field.name))
+                values.append(field.name)
+            csvwriter.writerow(values)
+
+            # Write the values
+            row_index = 0
+            for row, record in enumerate(cursor.fetchall()):
+                values = []
+                num_values = 0
+                for col, value in enumerate(record):
+                    if value:
+                        num_values += 1
+                    values.append(value)
```

Comment:
Take care not to miss `orientation=False` (or other legit boolean fields). Is num_values just a count of "real" values?

For what it's worth I started thinking about fixing the well_orientation=False issue... sounds like it is NOT stored this way in the legacy database so we are free to change it on our end without breaking replication.

Reply:
Good catch - will change to be explicit on "" and None